### PR TITLE
Fix a tooltip bug with the JSONEditor

### LINKED ISSUE #2358: JSONEditor popup bug
When editing metdata, if you switch the jsoneditor to `Code` the text pop up remains. If you swtch back to `Tree` another pop up remains. I'm not yet sure if this is an issue with the JSONEditor package or girder web. 
screenshot

### PR DESCRIPTION
**Pull request by brianhelba**
Fixes #2358

### CODE REVIEW

File: clients/web/src/views/widgets/MetadataWidget.js [hunk 1]
```diff
@@ -317,7 +317,16 @@ var JsonMetadatumEditWidget = MetadatumEditWidget.extend({
...
             modes: ['code', 'tree'],
-            onError: function () {
+            onModeChange: () => {
+                // JSONEditor removes tooltip-containing elements without destroying their
+                // Bootstrap-styled tooltips first. Since this callback is not called until after
+                // the original tooltip-containing element is destroyed, the only way to clean up is
+                // to remove the tooltip from DOM.
+                // When used in an external app with no tooltip styling enabled, this call should
+                // have no effect (assuming there's no other ".tooltip" elements.
+                $('.tooltip').remove();
```
Comment (zachmullen) [inline clients/web/src/views/widgets/MetadataWidget.js]: Should this be `this.$`?

Comment (brianhelba) [reply clients/web/src/views/widgets/MetadataWidget.js]: Unfortunately, no. The tooltip is attached directly below `#g-app-body-container`, which is much higher in the hierarchy than this widget's element. I was considering using the selector `'#g-app-body-container>.tooltip'), but I wanted to keep it general enough to also correct tooltip problems if this was reused in another external app.

**Comment by sutartmelson** [comment]
The popup still has weird behavior insofar as it doesn't disappear on its own, you must click somewhere first. But its definitely an improvement. 

As long as we're okay with that and you were already aware, looks good to me. I can approve it.

**Comment by brianhelba** [comment]
@girder/developers  @sutartmelson The fact that button tooltips remain until clicking somewhere else is a natural consequence of accepting the [Bootstrap default of `{trigger: 'hover focus'}`](https://getbootstrap.com/docs/3.3/javascript/#tooltips-options) when initializing our tooltips. JSONEditor makes its buttons focusable, so the click causes the focus to stay on the button (and the tooltip to pop up) until another click clears the focus.

If we want, we could easily initialize tooltips with `{trigger: 'hover'}`, but it would make them inaccessible to users who are not using a traditional pointing device (e.g. mobile, keyboard selection, etc.). I don't have any personal feelings on that tradeoff.

**Comment by brianhelba** [comment]
@sutartmelson @zachmullen Please test and try it now.

**Comment by brianhelba** [comment]
@zachmullen Agreed. I'm not yet sure whether to totally blame this on Bootstrap (not giving us a cleaner way to veto tooltip initialization and not cleaning up old tooltips) or partially blame the code at hand (in both the JSONEditor library, and the Backbone architecture of the `MetadataWidget`-related classes).

I've scoped the workaround into a private method, and I plan to revist this as part of a larger overhaul of the way that our Views are initialized / destroyed.

**Comment by zachmullen** [comment]
BTW, the candela test is failing with the same race condition -- is your earlier fix incorporated into this branch?

**Comment by brianhelba** [comment]
@zachmullen No, #2369 isn't merged yet. I wanted to give it a little time to cool off, in case @jeffbaumes or other @girder/developers want to review it.

**Comment by brianhelba** [comment]
@sutartmelson Does this fix all the issues you've found?

**Comment by sutartmelson** [comment]
@brianhelba Yup!

**Pull request merged**